cupy/_functional/vectorize.py

```diff
@@ -6,13 +6,10 @@
 
 
 def _get_input_type(arg):
-    if isinstance(arg, int):
-        return 'l'
-    if isinstance(arg, float):
-        return 'd'
-    if isinstance(arg, complex):
-        return 'D'
-    return arg.dtype.char
+    if hasattr(arg, 'dtype'):
+        return arg.dtype.char
+    else:
+        return numpy.dtype(type(arg)).char
```

Comment:
This line seems like a slower replacement of the existing one?

Reply:
That returns `l` on Linux, `q` on Widnows.